src/main/java/com/thealgorithms/datastructures/lists/DoublyLinkedList.java

```diff
@@ -49,19 +53,153 @@ public DoublyLinkedList(int[] array) {
+     */
+    public int value;
+    /**
+     * This points to the link in front of the new link
+     */
+    public Link next;
+    /**
+     * This points to the link behind the new link
+     */
+    public Link previous;
+
+    /**
+     * Constructor
+     *
+     * @param value Value of node
+     */
+    public Link(int value) {
+        this.value = value;
+    }
+
+    /**
+     * Displays the node
+     */
+    public void displayLink() {
+        System.out.print(value + " ");
+    }
+
+    /**
+     * Main Method
+     *
+     * @param args Command line arguments
+     */
+    public static void main(String args[]) {
+        DoublyLinkedList myList = new DoublyLinkedList();
+        LinkOperations linkOperations = new LinkOperations();
+        linkOperations.insertHead(13);
+        linkOperations.insertHead(7);
+        linkOperations.insertHead(10);
+        myList.display(); // <-- 10(head) <--> 7 <--> 13(tail) -->
+        myList.displayBackwards();
+
+        linkOperations.insertTail(11);
+        myList.display(); // <-- 10(head) <--> 7 <--> 13 <--> 11(tail) -->
+        myList.displayBackwards();
+
+        linkOperations.deleteTail();
+        myList.display(); // <-- 10(head) <--> 7 <--> 13(tail) -->
+        myList.displayBackwards();
+
+        linkOperations.delete(7);
+        myList.display(); // <-- 10(head) <--> 13(tail) -->
+        myList.displayBackwards();
+
+        linkOperations.insertOrdered(23);
+        linkOperations.insertOrdered(67);
+        linkOperations.insertOrdered(3);
+        myList.display(); // <-- 3(head) <--> 10 <--> 13 <--> 23 <--> 67(tail) -->
+        linkOperations.insertElementByIndex(5, 1);
+        myList.display(); // <-- 3(head) <--> 5 <--> 10 <--> 13 <--> 23 <--> 67(tail) -->
+        myList.displayBackwards();
+        linkOperations.reverse(); // <-- 67(head) <--> 23 <--> 13 <--> 10 <--> 5 <--> 3(tail) -->
+        myList.display();
+
+        linkOperations.clearList();
+        myList.display();
+        myList.displayBackwards();
+        linkOperations.insertHead(20);
+        myList.display();
+        myList.displayBackwards();
+    }
+}
+
+/*
+ * This class implements the operations of the Link nodes.
+ */
+class LinkOperations{
+    /**
+     * Head refers to the front of the list
+     */
+    private Link head;
+    /**
+     * Tail refers to the back of the list
+     */
+    private Link tail;
+
+    /**
+     * Size refers to the number of elements present in the list
+     */
+    private int size;
+
     /**
      * Insert an element at the head
      *
      * @param x Element to be inserted
      */
     public void insertHead(int x) {
         Link newLink = new Link(x); // Create a new link with a value attached to it
-        if (isEmpty()) // Set the first element added to be the tail
+        DoublyLinkedList doublyLinkedList = new DoublyLinkedList();
+        if (doublyLinkedList.isEmpty()) // Set the first element added to be the tail
```

Comment:
What is the point of this change?

Reply:
I understand your point. I have made one more commit where I have removed this unnecessary object creations. Thanks!